File: 69-Sqrtx/69-Sqrtx.py
# Last updated: 9/16/2025, 12:09:56 AM
class Solution:
    def mySqrt(self, x: int) -> int:
        if x<=1:
            return x
        i=1
        z=i*i
        while z<=x:
            i+=1
            z=i*i
        return i-1   

              
        

File: 69-Sqrtx/69-Sqrtx.cpp
// Last updated: 9/16/2025, 12:09:51 AM
class Solution {
public:
    int mySqrt(int x) {
        if (x < 2) return x;

        int low=1,high=x/2;
        
        while(low<high){
            int mid=low+(high-low+1)/2;
            
            long long square=1LL*mid*mid;
            if(square==x) return mid;
            else if(square<x){
                low=mid;
            }else{
                high=mid-1;
            }
        }
        return low;
    }
};